tests/test_sequence.py

```diff
@@ -137,6 +137,221 @@ def test_magnetic_field():
     assert np.all(seq3_.magnetic_field == np.array((1.0, 0.0, 0.0)))
 
 
+def test_switch_device():
```

Comment:
There's an overarching issue with these tests: you are just trying to cover all the lines, rather than checking behaviour. The objective of the unit tests is to ensure everything is working properly. This includes checking that errors and warning happen when they should (which you are doing), but also asserting that the outcomes are correct when things go without errors, which involves adding `assert` statements - you are missing this part.

Reply:
You're rigth